pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,289 @@
+	cmd.Flags().StringArrayVar(&c.gpus, GPUFlag, c.gpus, "Specify the list of vGPUs to passthrough. Can be provided multiple times.")
+	cmd.Flags().StringArrayVar(&c.hostDevices, HostDeviceFlag, c.hostDevices, "Specify list of HostDevices to passthrough. Can be provided multiple times.")
+
+	cmd.MarkFlagRequired(CPUFlag)
+	cmd.MarkFlagRequired(MemoryFlag)
+	return cmd
+}
+
+func withGPUs(c *createInstancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, param := range c.gpus {
+		gpu := GPU{}
+
+		err := params.Map(GPUFlag, param, &gpu)
+		if err != nil {
+			return err
+		}
+
+		if gpu.Name == "" {
+			return params.FlagErr(GPUFlag, NameErr)
+		}
+
+		if gpu.DeviceName == "" {
+			return params.FlagErr(GPUFlag, DeviceNameErr)
+		}
+
+		instancetypeSpec.GPUs = append(instancetypeSpec.GPUs, v1.GPU{Name: gpu.Name, DeviceName: gpu.DeviceName})
+	}
+
+	return nil
+}
+
+func withHostDevices(c *createInstancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, param := range c.hostDevices {
+		hostDevice := HostDevice{}
+
+		err := params.Map(HostDeviceFlag, param, &hostDevice)
+		if err != nil {
+			return err
+		}
+
+		if hostDevice.Name == "" {
+			return params.FlagErr(HostDeviceFlag, NameErr)
+		}
+
+		if hostDevice.DeviceName == "" {
+			return params.FlagErr(HostDeviceFlag, DeviceNameErr)
+		}
+
+		instancetypeSpec.HostDevices = append(instancetypeSpec.HostDevices, v1.HostDevice{Name: hostDevice.Name, DeviceName: hostDevice.DeviceName})
+	}
+
+	return nil
+}
+
+func withIOThreadsPolicy(c *createInstancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	var policy v1.IOThreadsPolicy
+
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		policy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		policy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, IOThreadErr)
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &policy
+	return nil
+}
+
+func defaultCreateInstancetype() createInstancetype {
+	return createInstancetype{
+		name: "instancetype-" + rand.String(5),
+	}
+}
+
+func (c *createInstancetype) usage() string {
+	return `  # Create a manifest for a ClusterInstancetype with a random name:
+  {{ProgramName}} create instancetype --cpu 2 --memory 256Mi
+  
+  # Create a manifest for a ClusterInstancetype with a specified name:
+  {{ProgramName}} create instancetype --name my-instancetype --cpu 2 --memory 256Mi
+
+  # Create a manifest for a ClusterInstancetype with a specified gpu:
+  {{ProgramName}} create instancetype --cpu 2 --memory 256Mi --gpu name:gpu1,devicename:nvidia
+  
+  # Create a manifest for a Instancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --namespaced --name my-instancetype --cpu 2 --memory 256Mi`
```

Comment:
An example of piping this into `kubectl` to actually create the resources would be useful.